provider/source/src/time_zones/convert.rs

```diff
@@ -407,67 +409,67 @@ macro_rules! long_short_impls {
-                                        .0
-                                        .iter()
-                                        .flat_map(move |(inner_key, place_or_region)| {
-                                            let iana = format!("{key}/{inner_key}");
-                                            match place_or_region {
-                                                LocationOrSubRegion::Location(place) => {
-                                                    match bcp47_tzid_data.get(&iana) {
-                                                        Some(bcp47) => place
-                                                            .$field
-                                                            .clone()
-                                                            .and_then(|zf| type_fallback(&zf).cloned())
-                                                            .map(|format| vec![(bcp47, format)])
-                                                            .unwrap_or_default(),
-                                                        None => panic!("Cannot find bcp47 for {iana:?}."),
-                                                    }
-                                                }
-                                                LocationOrSubRegion::SubRegion(region) => region
-                                                    .iter()
-                                                    .filter_map(|(inner_key, place)| {
-                                                        let iana = format!("{iana}/{inner_key}");
-                                                        match bcp47_tzid_data.get(&iana) {
-                                                            Some(bcp47) => place
-                                                                .$field
-                                                                .clone()
-                                                                .and_then(|zf| type_fallback(&zf).cloned())
-                                                                .map(|format| (bcp47, format)),
-                                                            None => {
-                                                                panic!("Cannot find bcp47 for {iana:?}.")
-                                                            }
-                                                        }
-                                                    })
-                                                    .collect::<Vec<_>>(),
-                                            }
-                                        })
+                                .filter_map(|(key, metazone)| {
+                                    metazone.$field.as_ref().and_then(type_fallback).and_then(
+                                        |format| {
+                                            Some((
+                                                meta_zone_id_data.get(key)?.clone(),
```

Comment:
Question: this one now drops metazones instead of panicking? And similar elsewhere?

Reply:
Yes